### PR TITLE
[REVIEW] Add support for `size` property in `DataFrame`/ `Series` / `Index`/ `MultiIndex`

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #5730

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -390,6 +390,13 @@ def ndim(self):
         """
         return 1
 
+    @property
+    def size(self):
+        """
+        Return the number of elements in the underlying data.
+        """
+        return len(self)
```
Comment (kkraus14) [inline python/cudf/cudf/core/series.py:398]: It looks like this should be implemented for DataFrame / MultiIndex / Index / Array as well, where maybe we push it to both `Frame` and `Column` classes?

Comment (galipremsagar) [reply python/cudf/cudf/core/series.py]: Added a `Frame.size` which is used by `DataFrame`/`Index` & `Series`. `MultiIndex` actually, so added a separate property for that.

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -49,6 +49,91 @@ def _mimic_inplace(self, result, inplace=False):
         else:
             return result
 
+    @property
+    def size(self):
```
Comment (kkraus14) [inline python/cudf/cudf/core/frame.py:53]: Can we add a similar method to `Column` as well? (in preparation for eventually exposing them as `cudf.array`

Comment (galipremsagar) [reply python/cudf/cudf/core/frame.py:53]: `Column` already has a `size` property

File: python/cudf/cudf/core/multiindex.py [hunk 3]
```diff
@@ -615,9 +616,10 @@ def __eq__(self, other):
     def is_contiguous(self):
         return True
 
+    @copy_docstring(Frame.size)
```
Comment (shwina) [inline python/cudf/cudf/core/multiindex.py:619]: Do we need to do this? Doesn't Sphinx use the docstring of `Frame` here automatically? Python/IPython `help` does that even without `copy_docstring`.

Comment (galipremsagar) [reply python/cudf/cudf/core/multiindex.py]: Yep, this was not needed - Removed it. Sphinx picks the docs the same way help does.

**Pull request merged**